types/gitignore-parser/index.d.ts

```diff
@@ -0,0 +1,28 @@
+// Type definitions for gitignore-parser 0.0
+// Project: https://github.com/codemix/gitignore-parser
+// Definitions by: Alex Nuccio <https://github.com/anuccio1>
+//                 Matt Schwartz <https://github.com/microsoftly>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/**
+ * Compile the given `.gitignore` content (not filename!)
+ * and return an object with `accepts`, `denies` and `maybe` methods.
+ * These methods each accepts a single filename and determines whether
+ * they are acceptable or unacceptable according to the `.gitignore` definition.
+ *
+ */
+export function compile(
+    content: string,
+): {
+    accepts(input: string): boolean;
+    denies(input: string): boolean;
+    maybe(input: string): boolean;
+};
+
+/**
+ * Parse the given `.gitignore` content and return an array
+ * containing two further arrays - positives and negatives.
+ * Each of these two arrays in turn contains two regexps, one
+ * strict and one for 'maybe'.
+ */
+export function parse(content: string): RegExp[][];
```

Comment:
I think we can be more explicit with the typings here. We know that a 2D array of size 2X2 of RegExp is being returned.

so
```typescript
export type GitignoreParseResultSet = [RegExp, RegExp]; // [strict, maybe]

export function parse(content: string): [GitignoreParseResultSet, GitignoreParseResultSet]; // [positive, negative]
```

Reply:
great suggestion